RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc

```diff
@@ -754,23 +761,28 @@ bool DAClusterizerInZT_vect::purge(vertex_t& y, track_t& tks, double& rho0, cons
   int nUnique = 0;
   double sump = 0;
 
-  std::vector<double> inverse_zsums(nt), arg_cache(nt), eik_cache(nt);
+  std::vector<double> inverse_zsums(nt), arg_cache(nt), eik_cache(nt), pcut_cache(nt);
   double* pinverse_zsums;
   double* parg_cache;
   double* peik_cache;
+  double* ppcut_cache;
   pinverse_zsums = inverse_zsums.data();
   parg_cache = arg_cache.data();
   peik_cache = eik_cache.data();
+  ppcut_cache = pcut_cache.data();
+#pragma GCC ivdep
   for (unsigned i = 0; i < nt; ++i) {
     inverse_zsums[i] = tks.Z_sum_ptr[i] > eps ? 1. / tks.Z_sum_ptr[i] : 0.0;
   }
-
+  const auto rhoconst = rho0 * local_exp(-beta * dzCutOff_ * dzCutOff_);
+#pragma GCC ivdep
   for (unsigned int k = 0; k < nv; ++k) {
-    nUnique = 0;
-    sump = 0;
+    const double pmax = y.pk_ptr[k] / (y.pk_ptr[k] + rhoconst);
+    ppcut_cache[k] = uniquetrkweight_ * pmax;
```

Comment:
is the first ```const double pmax``` needed?
it is not added in the InZ_vect.cc file.
Please make these parts as consistent as possible.

Reply:
It needed to be this way to vectorize.

Reply:
just to be sure, are you saying that
```C++
    const double pmax = y.pk_ptr[k] / (y.pk_ptr[k] + rhoconst);
    ppcut_cache[k] = uniquetrkweight_ * pmax;
```
vectorizes
while
```C++
    ppcut_cache[k] = uniquetrkweight_ * y.pk_ptr[k] / (y.pk_ptr[k] + rhoconst);
```
does not?

Reply:
I removed it since it has no impact on vectorization.